service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -95,12 +102,44 @@ public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseDataBefore = getCaseDataBefore(callbackRequest);
 
         if (isRemovingNewSDO(caseData, caseDataBefore)) {
```

Comment:
I feel like this can be simplified from a controller point of view, the `isRemoving...` and corresponding `getRemoved...` methods can be merged and then we can publish the events if the optional isn't empty

Reply:
The resultant methods could also be moved to the removed order service

Reply:
Even further you could have have the method in the service generate a list of events which the controller then just loops through and publishes